### PR TITLE
fix remove of regulating points

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java [hunk 1]
```diff
@@ -2378,6 +2492,78 @@ private Map<OwnerInfo, RegulationPointAttributes> innerGetRegulationPoints(Prepa
...
+    }
+
+    private <T extends IdentifiableAttributes> void insertRegulatingEquipmentsInto(UUID networkUuid, int variantNum,
+                                                                                   String equipmentId, Resource<T> resource,
+                                                                                   ResourceType type) {
+        try (var connection = dataSource.getConnection()) {
+            var preparedStmt = connection.prepareStatement(QueryCatalog.buildRegulatingEquipmentsForOneEquipmentQuery());
+            preparedStmt.setObject(1, networkUuid);
+            preparedStmt.setInt(2, variantNum);
+            preparedStmt.setObject(3, type.toString());
```
Comment (antoinebhs) [inline network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java:2539]:
```suggestion
            preparedStmt.setString(3, type.toString());
```

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java [hunk 2]
```diff
@@ -2378,6 +2492,78 @@ private Map<OwnerInfo, RegulationPointAttributes> innerGetRegulationPoints(Prepa
...
+            preparedStmt.setObject(4, equipmentId);
```
Comment (antoinebhs) [inline network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java:2540]:
```suggestion
            preparedStmt.setString(4, equipmentId);
```
?

Comment (antoinebhs) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java:2540]: same in other places in your code?

**Review by antoinebhs — CHANGES_REQUESTED** [review]
First review:
* fix duplicated calls to insertRegulating*()
* separate the refacto

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java [hunk 3]
```diff
@@ -1603,43 +1703,34 @@ public void deleteHvdcLine(UUID networkUuid, int variantNum, String hvdcLineId)
...
+        Map<OwnerInfo, Map<String, ResourceType>> regulatingEquipments = getRegulatingEquipments(networkUuid, variantNum, ResourceType.DANGLING_LINE);
+        danglingLines.forEach(danglingLine -> {
+            OwnerInfo ownerInfo = new OwnerInfo(danglingLine.getId(), ResourceType.DANGLING_LINE, networkUuid, variantNum);
+            danglingLine.getAttributes().setRegulatingEquipments(regulatingEquipments.get(ownerInfo));
+        });
 
-    public void deleteGround(UUID networkUuid, int variantNum, String groundId) {
-        deleteIdentifiable(networkUuid, variantNum, groundId, GROUND_TABLE);
+        return danglingLines;
     }
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java:1734]: ok but we wont be able to regulate on these equipments and I think it is not important

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java [hunk 4]
```diff
@@ -926,7 +931,10 @@ public void createGenerators(UUID networkUuid, List<Resource<GeneratorAttributes
 
     public Optional<Resource<GeneratorAttributes>> getGenerator(UUID networkUuid, int variantNum, String generatorId) {
         Optional<Resource<GeneratorAttributes>> generatorResource = getIdentifiable(networkUuid, variantNum, generatorId, mappings.getGeneratorMappings());
-        generatorResource.ifPresent(generatorAttributesResource -> insertRegulationPointIntoEquipment(networkUuid, variantNum, generatorId, generatorAttributesResource, ResourceType.GENERATOR));
+        generatorResource.ifPresent(generatorAttributesResource ->
+            insertRegulatingPointIntoEquipment(networkUuid, variantNum, generatorId, generatorAttributesResource, ResourceType.GENERATOR));
+        generatorResource.ifPresent(generatorAttributesResource ->
+            insertRegulatingEquipmentsInto(networkUuid, variantNum, generatorId, generatorAttributesResource, ResourceType.GENERATOR));
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java:937]: same comment on same lines as the previous comment

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java [hunk 5]
```diff
@@ -926,7 +931,10 @@ public void createGenerators(UUID networkUuid, List<Resource<GeneratorAttributes
 
     public Optional<Resource<GeneratorAttributes>> getGenerator(UUID networkUuid, int variantNum, String generatorId) {
         Optional<Resource<GeneratorAttributes>> generatorResource = getIdentifiable(networkUuid, variantNum, generatorId, mappings.getGeneratorMappings());
-        generatorResource.ifPresent(generatorAttributesResource -> insertRegulationPointIntoEquipment(networkUuid, variantNum, generatorId, generatorAttributesResource, ResourceType.GENERATOR));
+        generatorResource.ifPresent(generatorAttributesResource ->
+            insertRegulatingPointIntoEquipment(networkUuid, variantNum, generatorId, generatorAttributesResource, ResourceType.GENERATOR));
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java]: my mistake

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java [hunk 6]
```diff
@@ -1046,15 +1076,29 @@ public void createLoads(UUID networkUuid, List<Resource<LoadAttributes>> resourc
...
     }
 
     public List<Resource<LoadAttributes>> getLoads(UUID networkUuid, int variantNum) {
-        return getIdentifiables(networkUuid, variantNum, mappings.getLoadMappings());
+        List<Resource<LoadAttributes>> loads = getIdentifiables(networkUuid, variantNum, mappings.getLoadMappings());
+        Map<OwnerInfo, Map<String, ResourceType>> regulatingEquipments = getRegulatingEquipments(networkUuid, variantNum, ResourceType.LOAD);
+        loads.forEach(load -> {
+            OwnerInfo ownerInfo = new OwnerInfo(load.getId(), ResourceType.LOAD, networkUuid, variantNum);
+            load.getAttributes().setRegulatingEquipments(regulatingEquipments.get(ownerInfo));
+        });
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java]: done

File: network-store-server/src/main/resources/db/changelog/changesets/regulationPoint_regulatingtype_20240808T110000Z.sql [hunk 7]
```diff
@@ -0,0 +1,32 @@
+UPDATE regulationpoint r1
```
Comment (antoinebhs) [inline network-store-server/src/main/resources/db/changelog/changesets/regulationPoint_regulatingtype_20240808T110000Z.sql:1]: Have you tested this migration with real size db? And that it still works after migration? Migration times are OK?

**Review by antoinebhs — COMMENTED** [review]
Few comments left

File: pom.xml [hunk 8]
```diff
@@ -50,7 +50,7 @@
     <properties>
         <sirocco.version>1.0</sirocco.version>
         <!-- FIXME: powsybl-network-store modules'version is overloaded in the dependencies section.The overloads and this property below have to be removed at next powsybl-ws-dependencies.version upgrade -->
-        <powsybl-network-store.version>1.18.2</powsybl-network-store.version>
+        <powsybl-network-store.version>1.20.0-SNAPSHOT</powsybl-network-store.version>
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply pom.xml:53]: yes

File: network-store-server/src/main/resources/db/changelog/changesets/regulationPoint_regulatingtype_20240808T110000Z.sql (hunk 7 above)
Comment (EtienneLt) [reply network-store-server/src/main/resources/db/changelog/changesets/regulationPoint_regulatingtype_20240808T110000Z.sql:1]: I think it was tested

**Review by antoinebhs — COMMENTED** [review]
Does not work for 3WT?

**Review by antoinebhs — APPROVED** [review]
Code OK
Test OK

**Review by jonenst — APPROVED** [review]

**Pull request closed without merging**